cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -411,7 +424,6 @@ DatasetAdminOpResponse executeAdmin(DatasetId instance, String method) throws Ex
     Principal principal = authenticationContext.getPrincipal();
     switch (method) {
       case "exists":
-        ensureAccess(instance);
```

Comment:
Let's add a comment saying `ensureAccess` is happening above. Also what is the reason for moving the check up?